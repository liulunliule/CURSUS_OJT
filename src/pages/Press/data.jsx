import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchPressData = () => {
  const [news, setNews] = useState([]);
  const [releases, setReleases] = useState([]);

  useEffect(() => {
    const fetchNews = async () => {
      try {
        const response = await axios.get('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_news');
        setNews(response.data);
      } catch (error) {
        console.error("Error fetching news data:", error);
      }
    };

    const fetchReleases = async () => {
      try {
        const response = await axios.get('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_releases');
        setReleases(response.data);
      } catch (error) {
        console.error("Error fetching releases data:", error);
      }
    };

    fetchNews();
    fetchReleases();
  }, []);

  return { news, releases };
};