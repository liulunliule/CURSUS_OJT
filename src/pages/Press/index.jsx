import React, { useState, useEffect } from "react";
import "./index.scss";

const Press = () => {
  const [selected, setSelected] = useState("Press");
  const [news, setNews] = useState([]);
  const [releases, setReleases] = useState([]);

  useEffect(() => {
    const fetchNews = async () => {
      const response = await fetch('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_news');
      const data = await response.json();
      setNews(data);
    };

    const fetchReleases = async () => {
      const response = await fetch('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_releases');
      const data = await response.json();
      setReleases(data);
    };

    fetchNews();
    fetchReleases();
  }, []);

  const handleSelect = (item) => {
    if (item === "About") {
      window.location.href = "/secondLayout/about";
    } else if (item === "Company") {
      window.location.href = "/secondLayout/company";
    } else if (item === "Blog") {
      window.location.href = "/secondLayout/blog";
    } else if (item === "Careers") {
      window.location.href = "/secondLayout/careers";
    } else if (item === "Press") {
      window.location.href = "/secondLayout/press";
    }
    setSelected(item);
  };

  return (
    <div className="press-page">
      <header className="custom-header" style={{ backgroundColor: "#f8f8f8" }}>
        <nav className="custom-nav">
          <ul className="custom-ul">
            {["About", "Blog", "Company", "Careers", "Press"].map((item) => (
              <li
                key={item}
                className={`custom-li ${selected === item ? "selected" : ""}`}
                onClick={() => handleSelect(item)}
              >
                {item}
              </li>
            ))}
          </ul>
        </nav>
      </header>

      <section className="hero" style={{ backgroundColor: "#f8f8f8" }}>
        <h1>What others are saying</h1>
      </section>

      <div className="content">
        <div className="sidebar">
          <div className="follow-buttons">
            <button className="follow1">Follow our Facebook page</button>
            <button className="follow2">Follow our X page</button>
          </div>
          <a href="#" className="learn-more">
            Learn More
          </a>
          <a href="#" className="help-center">
            Cursus Help Center
          </a>
        </div>
        <div className="right-section">
          <div className="news-section">
            <h2>Cursus in the News</h2>
            <p>
              For media interviews and inquiries, please send an email to{" "}
              <a href="mailto:press@cursus.com">press@cursus.com</a>
            </p>
            <div className="news-items">
              {news.map((item) => (
                <div key={item.Press_News_ID} className="news-item">
                  <p>{item.Press_At}</p>
                  <h3>{item.Title}</h3>
                  <p>{item.Intro}</p>
                  <a href="#">Read More</a>
                </div>
              ))}
            </div>
            <a className="see-more" href="#">
              See More News
            </a>
          </div>
          <div className="releases-section">
            <h2>Press Releases</h2>
            <p>For Release from Cursus</p>
            <div className="release-items">
              {releases.map((item) => (
                <div key={item.Press_Releases_ID} className="release-item">
                  <p>{item.Press_At}</p>
                  <h3>{item.Title}</h3>
                </div>
              ))}
            </div>
            <a className="see-more" href="#">
              See More Press Releases
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Press;