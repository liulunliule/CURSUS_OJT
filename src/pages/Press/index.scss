.press-page {
    font-family: Arial, sans-serif;
    margin-top: 56px;
    color: #333;
    p {
      color: gray;
      margin-bottom: 30px;
    }
  }

  .custom-header {
    background: #f8f8f8;
    padding: 10px 0;
  }
  
  .custom-nav {
    .custom-ul {
      list-style: none;
      display: flex;
      justify-content: center;
      gap: 20px;
      padding: 0;
      margin: 0;
  
      .custom-li {
        cursor: pointer;
        padding: 10px;
        position: relative;
  
        &.selected::after {
          content: '';
          position: absolute;
          bottom: -3px;
          left: 0;
          right: 0;
          height: 3px;
          background: red;
        }
  
        &:hover {
          font-weight: bold;
        }
      }
    }
  }

    .hero {
      background: #f0f0f0;
      text-align: center;
      padding: 50px 0;
      font-weight: bold;
    }
    
    .hero h1 {
      font-size: 1.5em;
    }

    .content {
      background: #f0f0f0;
      display: flex;
      justify-content: space-between;
  
      .sidebar {
        width: 30%;
        padding-right: 20px;

        .follow-buttons {
          display: flex;
          flex-direction: column;
          align-items: center;
          gap: 10px;
          margin-top: 10px;
          margin-bottom: 45px;
        
          .follow1, .follow2 {
            display: flex;
            justify-content: center;
            align-items: center;
            padding: 10px;
            width: 50%;
            height: 30px;
            border: none;
            border-radius: 5px;
            cursor: pointer;
            color: #fff;
            font-size: 12px;
            font-weight: bold;
            text-align: center;
            transition: background-color 0.3s, transform 0.2s;
        
            &:hover {
              transform: scale(1.05);
            }
          }
        
          .follow1 {
            background: #007bff;
        
            &:hover {
              background: #0056b3;
            }
          }
        
          .follow2 {
            background: black;
        
            &:hover {
              background: #333;
            }
          }
        }

        .learn-more {
          color: gray;
          text-align: center;
          padding: 178px;
          text-decoration: none;
        }

        .help-center {
          display: block;
          text-align: center;
          padding: 10px;
          color: darkblue;
          text-decoration: none;
        }
      }
  
      .right-section {
        width: 70%;
  
        .news-section,
        .releases-section {
          margin-bottom: 40px;
  
          h2 {
            font-size: 1.5em;
            margin-bottom: 10px;
          }
  
          p {
            font-size: 1em;
            margin-bottom: 20px;
          }
  
          .news-items,
          .release-items {
            .news-item,
            .release-item {
              border: 1px solid #ddd;
              background: #f8f8f8;
              padding: 15px;
              border-radius: 5px;
              margin-bottom: 20px;
  
              p {
                margin: 0;
                font-size: 0.9em;
                color: #555;
              }
  
              h3 {
                margin: 5px 0;
                font-size: 1.2em;
                color: #333;
              }
  
              a {
                color: black;
                text-decoration: none;
                &:hover {
                  text-decoration: underline;
                }
              }
            }
          }
  
        .see-more {
          display: block;
          text-align: center;
          margin-top: 10px;
          color: black;
          text-decoration: none;
  
        &:hover {
          text-decoration: underline;
        }
      }
    }
  }
}  