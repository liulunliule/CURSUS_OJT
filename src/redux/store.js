import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
