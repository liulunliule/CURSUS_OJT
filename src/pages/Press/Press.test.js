import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from '../../redux/store';
import Press from '.';

test('test Press', () => {
  const tree = renderer.create(
    <Provider store={store}>
        <MemoryRouter>
            <Press/>
        </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});