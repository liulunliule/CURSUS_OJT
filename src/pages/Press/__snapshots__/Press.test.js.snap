// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Press 1`] = `
<div
  className="press-page"
>
  <header
    className="custom-header"
    style={
      Object {
        "backgroundColor": "#f8f8f8",
      }
    }
  >
    <nav
      className="custom-nav"
    >
      <ul
        className="custom-ul"
      >
        <li
          className="custom-li "
          onClick={[Function]}
        >
          About
        </li>
        <li
          className="custom-li "
          onClick={[Function]}
        >
          Blog
        </li>
        <li
          className="custom-li "
          onClick={[Function]}
        >
          Company
        </li>
        <li
          className="custom-li "
          onClick={[Function]}
        >
          Careers
        </li>
        <li
          className="custom-li selected"
          onClick={[Function]}
        >
          Press
        </li>
      </ul>
    </nav>
  </header>
  <section
    className="hero"
    style={
      Object {
        "backgroundColor": "#f8f8f8",
      }
    }
  >
    <h1>
      What others are saying
    </h1>
  </section>
  <div
    className="content"
  >
    <div
      className="sidebar"
    >
      <div
        className="follow-buttons"
      >
        <button
          className="follow1"
        >
          Follow our Facebook page
        </button>
        <button
          className="follow2"
        >
          Follow our X page
        </button>
      </div>
      <a
        className="learn-more"
        href="#"
      >
        Learn More
      </a>
      <a
        className="help-center"
        href="#"
      >
        Cursus Help Center
      </a>
    </div>
    <div
      className="right-section"
    >
      <div
        className="news-section"
      >
        <h2>
          Cursus in the News
        </h2>
        <p>
          For media interviews and inquiries, please send an email to
           
          <a
            href="mailto:press@cursus.com"
          >
            press@cursus.com
          </a>
        </p>
        <div
          className="news-items"
        />
        <a
          className="see-more"
          href="#"
        >
          See More News
        </a>
      </div>
      <div
        className="releases-section"
      >
        <h2>
          Press Releases
        </h2>
        <p>
          For Release from Cursus
        </p>
        <div
          className="release-items"
        />
        <a
          className="see-more"
          href="#"
        >
          See More Press Releases
        </a>
      </div>
    </div>
  </div>
</div>
`;